## **DSCI 504**

### **Week 8: Corporate Backbrief Supporting Queries**

**Student Name:** SEIF KUNGULIO

### **Assignment Overview:**

The Outdoor Performance Center (OPC) is evaluating the potential acquisition of **Ord Cycles**, a mountain bike manufacturer. As a data analyst on the OPC Analytics Team, I was tasked with determining whether this acquisition is a sound strategic move. This analysis uses data housed in **PostgreSQL** and queried via **Azure Data Studio** to assess business growth, operational capacity, fraud risks, and configuration preferences based on historical mountain bike sales data from 2001 to 2020.

### **Understanding PostgreSQL’s Role in Analytics:**

PostgreSQL played a critical role in enabling efficient data exploration and analysis. Its support for relational structures, advanced querying (e.g., window functions, CTEs), and data integrity mechanisms made it an ideal platform for evaluating historical trends and generating actionable insights. However, PostgreSQL's lack of built-in visualization tools and the need for query optimization skills are noteworthy constraints.

### **1\. Adequacy of Mountain Bike Sales for Growth**

To determine if OPC has sufficient momentum in mountain bike sales to justify growth, I analyzed yearly sales totals filtered by the "mountain" description. The SQL query below aggregates total order value by year:

In [1]:
-- -- Aggregates total order value per year for 'mountain' bikes.
SELECT DATE_PART('year', ord_date) AS year,
       SUM(order_tot) AS total_sales
FROM dsci_504.orders o
JOIN dsci_504.order_items oi ON o.ord_id = oi.ord_id
JOIN dsci_504.products p ON oi.prod_id = p.prod_id
WHERE LOWER(p.prod_description) ILIKE '%mountain%'
GROUP BY year
ORDER BY year;

(26 row(s) affected)

Total execution time: 00:00:02.219

year total_sales 2000 728573.98 2001 696021.07 2002 565160.48 2003 523462.78 2004 577862.23 2005 635804.33 2006 754774.44 2007 598244.74 2008 607583.88 2009 651427.80 2010 655375.09 2011 630545.52 2012 610428.02 2013 637818.16 2014 534454.30 2015 652318.08 2016 602895.41 2017 107852651.43 2018 101342881.33 2019 103331872.63 2020 251424173.58 2021 71582943.29 2022 106972242.64 2023 102113925.09 2024 115132846.82 2025 106696110.76

The query reveals annual revenue from mountain bike sales, showing a stable performance from 2000 to 2016 with revenues in the range of $500K–$750K. However, a significant surge appears from 2017 onward, with sales jumping to over $100 million per year and peaking at $251M in 2020. This sharp uptick likely indicates an increased consumer interest or business expansion, making a strong case for OPC's strategic growth and Ord Cycles' acquisition.  

Additionally, I evaluated product descriptions to identify top-performing bike types:

In [2]:
-- Identifies top mountain bike descriptions by total revenue and count of orders.
SELECT p.prod_description, COUNT(*) AS orders, SUM(oi.line_total) AS revenue
FROM dsci_504.order_items oi
JOIN dsci_504.products p ON oi.prod_id = p.prod_id
WHERE LOWER(p.prod_description) ILIKE '%mountain%'
GROUP BY p.prod_description
ORDER BY revenue DESC;

(2 row(s) affected)

Total execution time: 00:00:00.053

prod_description orders revenue Full-Suspension Mountain Bike 30857 293985124.14 Hardtail Mountain Bike 1894 8464768.19

Two mountain bike types dominate the sales:

- <span style="color: var(--vscode-foreground);">Full-Suspension Mountain Bike: $293M revenue from 30,857 orders.</span>
- Hardtail Mountain Bike: $8.4M from 1,894 orders

<span style="color: var(--vscode-foreground);">This suggests customers overwhelmingly prefer Full-Suspension builds, which should guide future inventory, marketing, and product alignment with Ord Cycles.</span>

### **2\. Fraud Risk Analysis**

To ensure due diligence, I evaluated potential fraud by identifying customers with a high frequency of returns and high refund totals. Customers flagged by the following query could indicate abusive return behavior:

In [3]:
-- Flags customers with more than 3 returns and refund total above $500.
SELECT c.cus_id, c.cus_first_name, c.cus_last_name, COUNT(*) AS return_count,
       SUM(r.return_amount) AS total_returned
FROM dsci_504.return_items r
JOIN dsci_504.returns ret ON r.rac_id = ret.rac_id
JOIN dsci_504.customers c ON ret.cus_id = c.cus_id
GROUP BY c.cus_id, c.cus_first_name, c.cus_last_name
HAVING COUNT(*) > 3 AND SUM(r.return_amount) > 500
ORDER BY total_returned DESC;

(640 row(s) affected)

Total execution time: 00:00:00.099

cus_id cus_first_name cus_last_name return_count total_returned 125 Harper Robinson 7 51041.00 1393 Erin Manning 7 48308.33 1821 Tristan Bangeson 6 48255.84 1997 Alister Connery 6 44394.00 2582 Carl Stevens 7 43977.08 1675 Lee Reynolds 7 43770.58 1648 Mads Thomason 7 43770.58 1967 Margo Brooks 5 42900.00 1163 Brian Hart 5 42240.00 373 Carl Saylor 7 42094.95 672 Cooper Aaronson 5 41291.20 1909 Farris Willis 5 41194.95 38 Harper Jackson 6 40835.20 1062 Valerie Manchester 6 40367.20 47 Brandon Manchester 6 40003.45 2070 Lee Presten 5 40000.00 1645 Aaron Seiler 5 40000.00 281 Robert Garth 6 39922.70 2108 Tristan Deleon 5 38499.95 306 Carl Garcia 5 38474.80 1058 Adam Ramirez 6 37694.64 466 Jasmine Garth 8 37432.00 697 Brian Connery 6 36995.00 439 Alister Manchester 7 36902.64 1552 Joshua Presley 7 36400.00 1728 Thomas Manning 6 35938.34 335 Albert Harper 8 35156.00 962 Eric Cooper 8 34754.02 709 Timothy Archibald 5 34519.00 964 Nadine Manchester 5 34343.95 2367 Chris Bangeson 5 34290.00 2517 Mads Presten 6 34056.80 674 George Jamieson 5 33971.35 272 Blain Saylor 4 33920.00 818 Airn Booker 6 33893.50 2071 Mike Acosta 6 33688.02 2285 Mindy Slain 5 33564.96 602 Maggs Reynolds 6 33563.25 1194 Zachary Willis 8 33511.60 56 Corey Garth 5 33280.00 271 Corey Tankersly 5 33043.96 1944 Joshua Brooks 7 33020.40 2554 Corey Hart 8 32775.12 731 Albert Johnasen 5 32695.76 1813 Eric Connery 4 32647.96 925 Miguel Etzler 4 32570.96 1087 Richard Saylor 5 32372.72 2290 Leah Reynolds 4 32262.96 1907 Joshua Acosta 4 32078.40 1109 Reginald MacBain 4 32002.76 414 Reginald Jones 4 32000.00 1101 Reginald Connery 6 31975.00 1238 Chris Calahan 5 31887.96 1753 Adam Slain 7 31800.00 1158 Mads Dawson 4 31700.12 976 Moritz MacBain 4 31700.12 781 Brian Calahan 5 31630.72 1679 Jasmine Stevens 7 31535.00 646 Manuel Bangeson 8 31534.93 2524 Brice Ramirez 4 31519.76 2253 Thomas Maurer 4 31371.76 1228 Cooper Stevens 5 31264.70 464 George Sherman 4 31253.36 1634 Mads Applewood 5 31204.95 168 Brandon Slain 5 31102.50 913 Mindy Gorman 7 31073.91 1257 Leah Smith 4 30913.04 538 Adam Ramirez 5 30836.20 2294 Kenadi Cooper 5 30662.20 1606 Leah Ellis 6 30358.80 732 Cooper Garth 5 30197.00 1623 Abigail Ridgley 5 30154.80 1254 Noel Peters 6 29646.18 144 Maggs Manning 6 29608.80 134 Manuel Jones 6 29548.50 1068 Kenadi Jackson 5 29493.75 1510 Brandon Thomason 5 29410.35 1353 Margo Rice 7 29281.75 2069 Aaron Rice 5 29240.95 456 Cooper Tompsen 5 29171.95 678 Leah Saylor 6 29097.00 328 Chris Aaronson 6 29093.76 1934 Mike Roberts 7 28730.14 728 Timothy Jones 5 28723.75 76 Mike Slain 5 28682.50 320 Eric Smith 5 28600.00 2260 Eric Delapena 5 28600.00 1479 Robert MacBain 6 28492.75 2183 Robert Willis 7 28307.58 149 Reginald Deleon 5 28271.05 1815 Corey Ellis 5 27875.65 2279 Thomas Presten 5 27782.15 1335 Brandon Smith 6 27649.80 95 Jasmine Manchester 6 27629.72 2592 Brandon Willis 5 27532.95 1739 Erin Peters 4 27475.16 2072 Leah Riochardson 4 27475.16 902 Taylor Roberts 4 27338.64 612 Abigail Delacruz 5 27279.65 2173 Cooper French 4 27114.80 516 Margo Delapena 5 27034.80 2430 Mads Ridgley 5 26879.95 2482 Samuel Rowley 4 26603.20 1223 Chris Conley 5 26558.80 1778 Jasmine Shelton 6 26520.00 2104 Harper Williamson 7 26465.58 53 Alister Dawson 6 26394.00 1924 Jack Jordan 5 26375.00 1513 Cooper Jones 5 25887.96 2424 Noel Harper 7 25876.58 2206 Reginald Rice 5 25852.80 2390 Samuel Smith 4 25616.00 2548 Margo Dickerson 7 25517.64 1837 Summer Applewood 6 25500.48 2293 Manuel MacBain 5 25483.80 2546 Thomas Reynolds 5 25465.00 2494 Sarah Garth 7 25442.64 644 Manuel Banks 6 25433.64 2540 Kristoff Thomason 7 25433.64 179 Thomas Saylor 6 25433.64 1969 Blain Saylor 7 25194.00 2164 Alister Presley 5 25148.50 1897 Abigail Harper 5 25145.00 1124 Farris Booker 5 25130.76 1120 Richard Dewey 4 25094.36 1286 Margo Cooper 5 25094.36 2516 Taylor Bangeson 5 25015.80 571 Jonathon Reynolds 4 25011.76 1990 Abigail Monk 5 25000.00 288 Mike Sherman 4 24952.76 2066 Thomas Reynolds 4 24917.36 598 Alister Aaronson 6 2467

The results flagged over 600 customers with high return counts and refund values exceeding $500. For example:

- <span style="color: var(--vscode-foreground);">Harper Robinson had 7 returns totaling over $51,000.</span>
- Erin Manning and Tristan Bangeson had similar patterns

<span style="color: var(--vscode-foreground);">Such behavior could indicate abuse of the return policy, resulting in potential financial loss and operational inefficiency. This insight supports implementing stricter return controls and monitoring, especially important during and after the acquisition.</span>

### **3\. Warehouse Capacity to Meet Future Demand**

I explored warehouse performance and logistical readiness to determine if OPC must scale operations to accommodate Ord Cycles. The number of orders shipped per month offers insight into fulfillment volume:

In [4]:
-- Summarizes number of orders shipped per month for logistics trend analysis.
SELECT DATE_TRUNC('month', ord_ship_date) AS month,
       COUNT(ord_id) AS orders_shipped
FROM dsci_504.orders
WHERE ord_ship_date IS NOT NULL
GROUP BY month
ORDER BY month;

(313 row(s) affected)

Total execution time: 00:00:00.061

month orders_shipped 2000-01-01 00:00:00-06 9 2000-02-01 00:00:00-06 8 2000-03-01 00:00:00-06 14 2000-04-01 00:00:00-06 12 2000-05-01 00:00:00-05 13 2000-06-01 00:00:00-05 16 2000-07-01 00:00:00-05 19 2000-08-01 00:00:00-05 14 2000-09-01 00:00:00-05 13 2000-10-01 00:00:00-05 14 2000-11-01 00:00:00-06 11 2000-12-01 00:00:00-06 6 2001-01-01 00:00:00-06 7 2001-02-01 00:00:00-06 7 2001-03-01 00:00:00-06 17 2001-04-01 00:00:00-06 13 2001-05-01 00:00:00-05 14 2001-06-01 00:00:00-05 14 2001-07-01 00:00:00-05 11 2001-08-01 00:00:00-05 12 2001-09-01 00:00:00-05 8 2001-10-01 00:00:00-05 13 2001-11-01 00:00:00-06 12 2001-12-01 00:00:00-06 9 2002-01-01 00:00:00-06 7 2002-02-01 00:00:00-06 10 2002-03-01 00:00:00-06 12 2002-04-01 00:00:00-06 9 2002-05-01 00:00:00-05 10 2002-06-01 00:00:00-05 6 2002-07-01 00:00:00-05 9 2002-08-01 00:00:00-05 12 2002-09-01 00:00:00-05 9 2002-10-01 00:00:00-05 12 2002-11-01 00:00:00-06 17 2002-12-01 00:00:00-06 11 2003-01-01 00:00:00-06 10 2003-02-01 00:00:00-06 6 2003-03-01 00:00:00-06 14 2003-04-01 00:00:00-06 9 2003-05-01 00:00:00-05 8 2003-06-01 00:00:00-05 16 2003-07-01 00:00:00-05 7 2003-08-01 00:00:00-05 10 2003-09-01 00:00:00-05 5 2003-10-01 00:00:00-05 9 2003-11-01 00:00:00-06 6 2003-12-01 00:00:00-06 11 2004-01-01 00:00:00-06 8 2004-02-01 00:00:00-06 7 2004-03-01 00:00:00-06 10 2004-04-01 00:00:00-06 9 2004-05-01 00:00:00-05 11 2004-06-01 00:00:00-05 10 2004-07-01 00:00:00-05 15 2004-08-01 00:00:00-05 5 2004-09-01 00:00:00-05 9 2004-10-01 00:00:00-05 8 2004-11-01 00:00:00-06 12 2004-12-01 00:00:00-06 15 2005-01-01 00:00:00-06 12 2005-02-01 00:00:00-06 7 2005-03-01 00:00:00-06 9 2005-04-01 00:00:00-06 13 2005-05-01 00:00:00-05 10 2005-06-01 00:00:00-05 3 2005-07-01 00:00:00-05 17 2005-08-01 00:00:00-05 11 2005-09-01 00:00:00-05 11 2005-10-01 00:00:00-05 11 2005-11-01 00:00:00-06 17 2005-12-01 00:00:00-06 8 2006-01-01 00:00:00-06 12 2006-02-01 00:00:00-06 13 2006-03-01 00:00:00-06 12 2006-04-01 00:00:00-06 9 2006-05-01 00:00:00-05 9 2006-06-01 00:00:00-05 14 2006-07-01 00:00:00-05 18 2006-08-01 00:00:00-05 17 2006-09-01 00:00:00-05 14 2006-10-01 00:00:00-05 18 2006-11-01 00:00:00-06 12 2006-12-01 00:00:00-06 8 2007-01-01 00:00:00-06 8 2007-02-01 00:00:00-06 10 2007-03-01 00:00:00-06 7 2007-04-01 00:00:00-05 12 2007-05-01 00:00:00-05 8 2007-06-01 00:00:00-05 14 2007-07-01 00:00:00-05 8 2007-08-01 00:00:00-05 7 2007-09-01 00:00:00-05 5 2007-10-01 00:00:00-05 13 2007-11-01 00:00:00-05 16 2007-12-01 00:00:00-06 17 2008-01-01 00:00:00-06 13 2008-02-01 00:00:00-06 6 2008-03-01 00:00:00-06 12 2008-04-01 00:00:00-05 10 2008-05-01 00:00:00-05 8 2008-06-01 00:00:00-05 8 2008-07-01 00:00:00-05 16 2008-08-01 00:00:00-05 10 2008-09-01 00:00:00-05 12 2008-10-01 00:00:00-05 7 2008-11-01 00:00:00-05 12 2008-12-01 00:00:00-06 6 2009-01-01 00:00:00-06 11 2009-02-01 00:00:00-06 7 2009-03-01 00:00:00-06 12 2009-04-01 00:00:00-05 11 2009-05-01 00:00:00-05 14 2009-06-01 00:00:00-05 14 2009-07-01 00:00:00-05 3 2009-08-01 00:00:00-05 15 2009-09-01 00:00:00-05 9 2009-10-01 00:00:00-05 14 2009-11-01 00:00:00-05 8 2009-12-01 00:00:00-06 12 2010-01-01 00:00:00-06 13 2010-02-01 00:00:00-06 6 2010-03-01 00:00:00-06 11 2010-04-01 00:00:00-05 11 2010-05-01 00:00:00-05 6 2010-06-01 00:00:00-05 11 2010-07-01 00:00:00-05 13 2010-08-01 00:00:00-05 9 2010-09-01 00:00:00-05 15 2010-10-01 00:00:00-05 17 2010-11-01 00:00:00-05 8 2010-12-01 00:00:00-06 16 2011-01-01 00:00:00-06 15 2011-02-01 00:00:00-06 9 2011-03-01 00:00:00-06 9 2011-04-01 00:00:00-05 11 2011-05-01 00:00:00-05 15 2011-06-01 00:00:00-05 11 2011-07-01 00:00:00-05 9 2011-08-01 00:00:00-05 10 2011-09-01 00:00:00-05 9 2011-10-01 00:00:00-05 10 2011-11-01 00:00:00-05 8 2011-12-01 00:00:00-06 16 2012-01-01 00:00:00-06 9 2012-02-01 00:00:00-06 11 2012-03-01 00:00:00-06 12 2012-04-01 00:00:00-05 13 2012-05-01 00:00:00-05 10 2012-06-01 00:00:00-05 9 2012-07-01 00:00:00-05 6 2012-08-01 00:00:00-05 8 2012-09-01 00:00:00-05 9 2012-10-01 00:00:00-05 13 2012-11-01 00:00:00-05 18 2012-12-01

Order fulfillment shows a clear growth trajectory:

- <span style="color: var(--vscode-foreground);">From 2000 to 2016: monthly shipping volumes hover between 5 to 20 orders.</span>
- In 2017 and beyond: a surge to 90–220+ shipments per month, peaking in 2020

<span style="color: var(--vscode-foreground);">This pattern suggests an increasing strain on warehouse operations, reinforcing the need to assess and potentially expand fulfillment infrastructure to meet growing demand, especially post-acquisition.</span>  

Warehouse order distribution was also assessed:

In [5]:
-- Counts total orders processed by each warehouse.
SELECT w.warehouse_name, COUNT(*) AS total_orders
FROM dsci_504.orders o
JOIN dsci_504.warehouses w ON o.warehouse_id = w.warehouse_id
GROUP BY w.warehouse_name
ORDER BY total_orders DESC;

(3 row(s) affected)

Total execution time: 00:00:00.033

warehouse_name total_orders Sacramento 4251 Dallas 4243 Columbus 4105

Order processing is evenly split across three main warehouses:

- <span style="color: var(--vscode-foreground);">Sacramento: 4,251 orders.</span>
- Dallas: 4,243 orders.
- Columbus: 4,105 orders.

<span style="color: var(--vscode-foreground);">Although the distribution is relatively balanced, the high volumes indicate that any further load due to Ord Cycles could require logistical scaling or redistribution strategies to prevent bottlenecks.</span>

**4\. Preferred Mountain Bike Configurations**

Understanding which bike configurations are most popular helps align Ord Cycles' product offerings with customer preferences. The following query identifies the most frequently built configurations based on component count:

In [6]:
-- Determines most popular bike builds based on component count and usage.
SELECT b.build_name, COUNT(DISTINCT bc.comp_id) AS component_count, COUNT(*) AS popularity
FROM dsci_504.builds b
JOIN dsci_504.build_components bc ON b.build_id = bc.build_id
GROUP BY b.build_name
ORDER BY popularity DESC
LIMIT 5;

(5 row(s) affected)

Total execution time: 00:00:00.259

build_name component_count popularity wild ride 11 11 pinnacle 10 10 bull 10 10 loam assault 10 10 max traction 10 10

The most popular bike builds include:

- <span style="color: var(--vscode-foreground);">Wild Ride (11 components, 11 builds).</span>
- Pinnacle, Bull, Loam Assault, Max Traction (10 components each, 10 builds).

<span style="color: var(--vscode-foreground);">These configurations reflect consumer preferences, and serve as a guide for product development alignment post-acquisition. Ord Cycles should consider focusing on builds with 10–11 components, which are trending in OPC’s history.</span>

### **5\. Projected Growth Over the Next Quarter**

To anticipate demand in the near term, I used the average of recent quarters' sales as a forecast for upcoming growth:

In [7]:
-- Retrieve total order values grouped by quarter for short-term sales forecasting.
SELECT DATE_TRUNC('quarter', ord_date) AS quarter,
       SUM(order_tot) AS total_sales
FROM dsci_504.orders
GROUP BY quarter
ORDER BY quarter DESC
LIMIT 4;

(4 row(s) affected)

Total execution time: 00:00:00.022

quarter total_sales 2025-10-01 00:00:00-05 7223014.85 2025-07-01 00:00:00-05 7216169.54 2025-04-01 00:00:00-05 8633167.27 2025-01-01 00:00:00-06 6228300.81

Recent quarterly revenues:

- <span style="color: var(--vscode-foreground);">Q1 2025: <b>$6.2M</b></span>
- Q2 2025: **$8.6M**
- Q3 2025: **$7.2M**
- Q4 2025: **$7.2M**

<span style="color: var(--vscode-foreground);">The trend shows steady revenue in the <b>$6M–$8M</b> range, confirming stable short-term demand. This forecast supports investment decisions in operations, warehouse scaling, and marketing, ensuring readiness to integrate Ord Cycles.</span>

### **Strengths and Weaknesses of PostgreSQL for OPC Analytics:**

**Strengths of PostgreSQL for OPC Analytics**

1. <span style="color: var(--vscode-foreground);"><b>Robust Analytical Capabilities:&nbsp;</b></span> PostgreSQL supports advanced SQL constructs like Common Table Expressions (CTEs) and Window Functions, which allowed for complex analytical tasks (e.g., ranking builds, quarterly trend analysis, customer behavior segmentation) without requiring external tools.
2. **High Data Integrity and Structured Storage:** The relational schema (e.g., \`orders\`, \`products\`, \`returns\`) ensures consistency through primary/foreign key relationships—ideal for maintaining business-critical sales and customer data.
3. **Efficient for Large Volumes of Data:** Even with millions of rows (as seen with order and return logs), PostgreSQL provided sub-second query execution, supporting near real-time data exploration in Azure Data Studio.
4. **Customizable and Open Source:** Being open-source, PostgreSQL enables cost-effective deployment without licensing fees and is highly customizable to fit into existing infrastructure—especially relevant for a growing company like OPC.
5. **Compatible with External BI Tools:** PostgreSQL integrates well with Power BI, Tableau, and other dashboarding tools, allowing analysts to perform modeling in SQL and visualize trends through external applications.

**Weaknesses of PostgreSQL for OPC Analytics**

1. <span style="color: var(--vscode-foreground);"><b>Lacks Built-In Visualization:</b>&nbsp;</span> PostgreSQL doesn’t offer built-in dashboard or reporting interfaces. As a result, analysts must export results to external BI platforms, adding friction to executive reporting and real-time decision-making.
2. **Steep Learning Curve for Non-Technical Users:** Crafting performant SQL queries (e.g., JOINs, aggregations, filtering logic) requires specialized SQL knowledge. This limits accessibility for business users or junior analysts unfamiliar with database operations.
3. **Manual ETL and Orchestration Needs:** PostgreSQL does not natively support automated data pipelines, scheduling, or transformation orchestration. This limits its usability for complex workflows unless paired with tools like Apache Airflow, dbt, or Python-based ETL scripts.
4. **Limited Native Support for Unstructured or Streaming Data:** PostgreSQL handles structured/tabular data very well, but JSON, arrays, and time-series data require additional configuration or extensions (like TimescaleDB), making it less versatile for unstructured or real-time analytics use cases.

### **Strategic Implication for OPC**

- PostgreSQL is a solid backbone for analytics involving structured sales, customer, and operational data.
- For maximum impact, OPC should pair PostgreSQL with BI tools (like Power BI) and adopt lightweight orchestration tools to automate workflows.
- As OPC scales post-acquisition, investing in training for analysts, query optimization practices, and a unified data strategy will unlock the full potential of PostgreSQL.

### **Convert the file to html:**

In [1]:
!jupyter nbconvert --to html DSCI504_Wk8_Backbrief_SKungulio.ipynb

[NbConvertApp] Converting notebook DSCI504_Wk8_Backbrief_SKungulio.ipynb to html


[NbConvertApp] Writing 381230 bytes to DSCI504_Wk8_Backbrief_SKungulio.html
